src/internal/text/TextAs.tsx

```diff
@@ -0,0 +1,25 @@
+import { createElement } from 'react';
+import { cn } from '../../utils/cn';
+import { textColorMap } from './textColorMap';
+import type { TextReact } from './types';
+
+type TextAsReact = {
+  className: string;
+  as: 'span' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+} & TextReact;
+
+export function TextAs({
+  as,
+  children,
+  className,
+  variant = 'foreground',
+  ...props
+}: TextAsReact) {
+  const textColor = textColorMap[variant];
```

Comment:
Kind of confused why we use the variant to switch `textColor` versus just passing the color through the `className`? I can see why we'd want to make an association between high-level emphasis names and color styling, but I feel like we should start with the simplest approach first and then refactor to find more optimal abstractions. Jumping to an abstraction beyond classNames introduces more vocab all devs need to keep in context.

I've never built a component library, but I urge us to second check any abstractions that do not have an absolute necessity. Could you explain to me where effort this saved by internal and external devs by using this variant system in comparison to just taking colors via className?

Reply:
`Text` components will never be exposed directly to the end users. These are internal components.

We want to display all possible values for color in a given `Text` component. Right now, you will get an autocompletion dialog when modify any `Text` component.

Sure, we could just have `className` but then you will need to remember all the possible class names specifically associated with text colors and that's a pain.

Reply:
Btw, the `className` will be added in the follow up when I fix up the Identity components.

Reply:
My main point is that instead of remembering a className, I have to remember a variant name so to me it's no saved effort and therefore an abstraction that doesn't need to exist. I'm all for easing required context for developers, but I feel like this doesn't achieve that.

Reply:
`variant name` are a common practice of Design Systems. As when building large project where you need to collaborate with many designers and engs, it's easier to align on `variant name` that atomic classnames.

A `variant name` can last years, a `classname` can last only a few months, till a new design decision will happen.